tensorflow_io/core/python/experimental/avro_record_dataset_ops.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import tensorflow as tf
+from tensorflow.python.data.ops import dataset_ops
+from tensorflow.python.data.ops.readers import _create_or_validate_filenames_dataset, _create_dataset_reader
+from tensorflow.python.data.util import convert
+from tensorflow_io.core.python.ops import core_ops
+
+_DEFAULT_READER_BUFFER_SIZE_BYTES = 256 * 1024  # 256 KB
+_DEFAULT_READER_SCHEMA = ""
+# From https://github.com/tensorflow/tensorflow/blob/v2.0.0/tensorflow/python/data/ops/readers.py
+
+class _AvroRecordDataset(dataset_ops.DatasetSource):
+    """A `Dataset` comprising records from one or more AvroRecord files."""
+
+    def __init__(self, filenames, buffer_size=None, reader_schema=None):
+        """Creates a `AvroRecordDataset`.
+        Args:
+          filenames: A `tf.string` tensor containing one or more filenames.
+          buffer_size: (Optional.) A `tf.int64` scalar representing the number of
+            bytes in the read buffer. 0 means no buffering.
+          reader_schema: (Optional.) A `tf.string` scalar representing the reader schema or None
+        """
+        self._filenames = filenames
+        self._buffer_size = convert.optional_param_to_tensor(
+            "buffer_size",
+            buffer_size,
+            argument_default=_DEFAULT_READER_BUFFER_SIZE_BYTES)
+        self._reader_schema = convert.optional_param_to_tensor(
+            "reader_schema",
+            reader_schema,
+            argument_default=_DEFAULT_READER_SCHEMA,
+            argument_dtype=tf.dtypes.string)
+        variant_tensor = core_ops.io_avro_record_dataset(self._filenames,
+                                                         self._buffer_size,
+                                                         self._reader_schema)
+        super(_AvroRecordDataset, self).__init__(variant_tensor)
+
+    @property
+    def element_spec(self):
+        return tf.TensorSpec([], tf.dtypes.string)
+
+
+class AvroRecordDataset(dataset_ops.DatasetV2):
```

Comment:
`dataset_ops.DatasetV2` is really `tf.data.Dataset`

Reply:
thanks, updated